src/names.jl

```diff
@@ -152,3 +152,43 @@ function named_axiskeys(A::AbstractArray)
     NT = NamedTuple{dimnames(A)}
     NT(axiskeys(A))
 end
+
+
+"""
+    rekey(A, (1:10, [:a, :b]))
+    rekey(A, 2 => [:a, :b])
+    rekey(A, :y => [:a, :b])
+    rekey(A, :y => q => [:a, :b])
+
+Rekey a KeyedArray via `Tuple`s, `dim => newkey`. If `A` also has named dimensions then you
+can also pass `dimname => newkey`, or even `oldname => newname => newkey` to both `rename`
+and `rekey` the specified dimension.
+"""
+rekey(x::AbstractArray, k2::Tuple) = KeyedArray(keyless(x), k2)
+function rekey(x::AbstractArray, k2::Pair{<:Integer, <:AbstractVector}...)
+    dims, vals = first.(k2), last.(k2)
+    new_key = map(enumerate(axiskeys(x))) do (i, k)
+        idx = findfirst(==(i), dims)
+        idx === nothing ? k : vals[idx]
+    end
+    return rekey(x, Tuple(new_key))
```

Comment:
While nothing is really zero cost, nice to avoid an Array -> Tuple where possible:
```suggestion
    new_keys = ntuple(ndims(x)) do d
        idx = findfirst(==(d), dims)
        idx === nothing ? axiskeys(x,d) : vals[idx]
    end
    return rekey(x, new_keys)
```